core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -141,6 +141,16 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
     }
   )
 
+  newGauge(
+    "yammer-metrics-total",
+    new Gauge[Int] {
+      def value = {
+        import scala.collection.JavaConverters._
+        com.yammer.metrics.Metrics.defaultRegistry().groupedMetrics().asScala.foldLeft(0)(_ + _._2.size())
```

Comment:
`groupedMetrics` creates a new `TreeMap` on every invocation. Also, the conversion makes yet another copy of these in scala. Can you instead just do `defaultRegistry().allMetrics().size()`? That does not create any unnecessary intermediate copies.